test/functional/lua/semver_spec.lua

```diff
@@ -0,0 +1,303 @@
+            version = '1.2.3-rc1+build.15',
+            want = {
+              major = 1,
+              minor = 2,
+              patch = 3,
+              prerelease = 'rc1',
+              build = 'build.15',
+            },
+          },
+        }
+        for _, tc in ipairs(testcases) do
+          it(
+            string.format('returns correct table for %q: version = %q', tc.desc, tc.version),
+            function()
+              assert.same(tc.want, semver.parse(tc.version, { strict = true }))
+            end
+          )
+        end
+      end)
+
+      describe('strict = false', function()
+        local testcases = {
+          {
+            desc = 'a version missing patch version',
+            version = '1.2',
+            want = { major = 1, minor = 2, patch = 0 },
+          },
+          {
+            desc = 'a version missing minor and patch version',
+            version = '1',
+            want = { major = 1, minor = 0, patch = 0 },
+          },
+          {
+            desc = 'a version missing patch version with prerelease',
+            version = '1.1-0',
+            want = { major = 1, minor = 1, patch = 0, prerelease = '0' },
+          },
+          {
+            desc = 'a version missing minor and patch version with prerelease',
+            version = '1-1.0',
+            want = { major = 1, minor = 0, patch = 0, prerelease = '1.0' },
+          },
+        }
+        for _, tc in ipairs(testcases) do
+          it(
+            string.format('returns correct table for %q: version = %q', tc.desc, tc.version),
+            function()
+              assert.same(tc.want, semver.parse(tc.version, { strict = false }))
+            end
+          )
+        end
+      end)
+    end)
+
+    describe('errors', function()
+      local testcases = {
+        { desc = 'a word', version = 'foo' },
+        { desc = 'trailing period character', version = '0.0.0.' },
+        { desc = 'leading period character', version = '.0.0.0' },
+        { desc = 'an empty string', version = '' },
+        { desc = 'negative major version', version = '-1.0.0' },
+        { desc = 'negative minor version', version = '0.-1.0' },
+        { desc = 'negative patch version', version = '0.0.-1' },
+        { desc = 'no parameters' },
+        { desc = 'nil', version = nil },
+        { desc = 'a number', version = 0 },
+        { desc = 'a float', version = 0.01 },
+        { desc = 'a table', version = {} },
+        { desc = 'leading invalid string', version = 'foobar1.2.3' },
+        { desc = 'trailing invalid string', version = '1.2.3foobar' },
+        { desc = 'an invalid prerelease', version = '1.2.3-%?' },
+        { desc = 'an invalid build', version = '1.2.3+%?' },
+        { desc = 'build metadata before prerelease', version = '1.2.3+build.0-rc1' },
+      }
+      for _, tc in ipairs(testcases) do
+        it(string.format('returns error for %s: version = %s', tc.desc, tc.version), function()
```

Comment:
```suggestion
        it(string.format('returns error for %s: version = %s', tc.desc, tostring(tc.version)), function()
```

Reply:
Interesting, why is this test error happening on CI but not when I run locally using `TEST_FILE=test/functional/lua/semver_spec.lua make functionaltest`? Is there a command I can run to reproduce this locally?

Reply:
Try changing `functionaltest` to `functionaltest-lua`

Reply:
Still passing when I run `TEST_FILE=test/functional/lua/semver_spec.lua make functionaltest-lua`.